System/Vector.cpp

```diff
@@ -51,6 +51,22 @@ namespace RTE {
 		return *this;
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	Vector & Vector::ClampMagnitude(float upperLimit, float lowerLimit) {
+		if (upperLimit < lowerLimit) {
+			std::swap(upperLimit, lowerLimit);
+		}
+		if (upperLimit == 0 && lowerLimit == 0) {
```

Comment:
What's handled in the next statement, the upperLimit and lowerLimit being 0?
If that's what you're talking about, the elses below both use GetMagnitude, which has to do various math, so I put this in here to make this case trivial and fast. That said, I think I'll put it right at the top of the method and early return instead of having it in this if-else, since it makes more sense there to me.

Reply:
Sounds good. Yes, I meant you could delete this case completely, but if you say it should be faster this way then 👍 .